packages/json-schema/lib/main.tsp

```diff
@@ -109,6 +109,18 @@ extern dec contentMediaType(target: string | Reflection.ModelProperty, value: va
  */
 extern dec contentSchema(target: string | Reflection.ModelProperty, value: unknown);
 
+/**
+ * Specify a custom property to add to the emitted schema. Useful for adding custom keywords
+ * and other vendor-specific extensions. The value will be converted to a schema unless the parameter
+ * is wrapped in the `Json<T>` template. For example, `@extension("x-schema", { x: "value" })` will
+ * emit a JSON schema value for `x-schema`, whereas `@extension("x-schema", Json<{x: "value"}>)` will
```

Comment:
I think this would be useful for OpenAPI as well, FWIW. I could swear it even came up before but I'm failing to find the issue. Maybe since we plan to have an object literal syntax and nobody is screaming for this in OpenAPI we don't need to move this to core or change OpenAPI extension now, though. I guess that means less impact from the inevitable deprecation.

Reply:
Oh I forgot that OpenAPI did the jsonification always. That's why this hasn't come up in a while. :(